src/pages/SignUp.tsx

```diff
@@ -1,7 +1,66 @@
-import React from 'react';
+import React from "react";
+import { useForm } from "react-hook-form";
+import { FaUserAlt, FaTwitterSquare } from "react-icons/fa";
+import { HiMail } from "react-icons/hi";
+import { RiLockPasswordFill } from "react-icons/ri";
+import { Link } from "react-router-dom";
+import { Container } from "../styles/Container";
+import { Section } from "../styles/Section";
+import { AuthField } from "../styles/AuthField";
+import { AuthButton } from "../styles/AuthButton";
 
-function Signup() {
-  return <div>Signup</div>;
-}
+const SignUp = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = (data: any) => console.log(data);
+  console.log(errors);
 
-export default Signup;
\ No newline at end of file
+  return (
+    <Container>
+      <Section>
+        <h1>Sign Up</h1>
+        <FaTwitterSquare className="twitterIcon" />
+        <h2>or use your email to register</h2>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <AuthField>
+            <FaUserAlt className="fieldIcon" />
+            <input
+              type="text"
+              placeholder="name"
+              {...register("name", { required: true, maxLength: 80 })}
+            />
+          </AuthField>
+
+          <AuthField>
+            <HiMail className="fieldIcon" />
+            <input
+              type="email"
+              placeholder="email"
+              {...register("email", { required: true, pattern: /^\S+@\S+$/i })}
+            />
+          </AuthField>
+
+          <AuthField>
+            <RiLockPasswordFill className="fieldIcon" />
+            <input
+              type="password"
+              placeholder="password"
+              {...register("password", { required: true })}
+            />
+          </AuthField>
+
+          <AuthButton type="submit">SIGN UP</AuthButton>
+        </form>
+        <h3>Already have an account?</h3>
+        <Link className="link" to="/sign-in">
+          Login Here
+        </Link>
```

Comment:
you can have this as a separate styled component
```
const StyledLink = styled(Link)`
   color: ....
   text-decoration: none
`;
```
and this will be
```
<StyledLink to="/sign-in"> Login Here </StyledLink>
```